src/ui/components/screensaver.c

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "screensaver.h"
+
+#include "image.h"
+#include "ui_images.h"
+
+#include <hardfault.h>
+#include <screen.h>
+#include <ui/ui_util.h>
+
+#include <string.h>
+
+static void _render(component_t* component)
+{
+    component_t* image = component->sub_components.sub_components[0];
+
+    static uint16_t counter = 0;
```

Comment:
shouldn't there be a `#include <stdint.h>` at the top for these types?

Reply:
done